Templates/make_jenkins_EC2-Agent-Windows-instance.tmplt.json

```diff
@@ -0,0 +1,1613 @@
+            "Type": "String"
+        },
+        "WatchmakerEnvironment": {
+            "AllowedValues": [
+                "",
+                "dev",
+                "test",
+                "prod"
+            ],
+            "Default": "",
+            "Description": "Environment in which the instance is being deployed",
+            "Type": "String"
+        },
+        "WatchmakerOuPath": {
+            "AllowedPattern": "^$|^(OU=.+,)+(DC=.+)+$",
+            "Default": "",
+            "Description": "(Optional) DN of the OU to place the instance when joining a domain. If blank and \"WatchmakerEnvironment\" enforces a domain join, the instance will be placed in a default container. Leave blank if not joining a domain, or if \"WatchmakerEnvironment\" is \"$false\"",
+            "Type": "String"
+        }
+    },
+    "Resources": {
+        "Ec2Res": {
+            "CreationPolicy": {
+                "ResourceSignal": {
+                    "Count": "1",
+                    "Timeout": "PT60M"
+                }
+            },
+            "Metadata": {
+                "AWS::CloudFormation::Init": {
+                    "configSets": {
+                        "appscript": [
+                            "reboot",
+                            "make-app"
+                        ],
+                        "chainscript": [
+                            "chain-script",
+                            "reboot"
+                        ],
+                        "launch": [
+                            "setup",
+                            {
+                                "Fn::If": [
+                                    "InstallCloudWatchAgent",
+                                    "install-cloudwatch-agent",
+                                    {
+                                        "Ref": "AWS::NoValue"
+                                    }
+                                ]
+                            },
+                            "watchmaker-launch",
+                            {
+                                "Fn::If": [
+                                    "ExecuteAppScript",
+                                    {
+                                        "ConfigSet": "appscript"
+                                    },
+                                    {
+                                        "Ref": "AWS::NoValue"
+                                    }
+                                ]
+                            },
+                            {
+                                "Fn::If": [
+                                    "Reboot",
+                                    "reboot",
+                                    {
+                                        "Ref": "AWS::NoValue"
+                                    }
+                                ]
+                            },
+                            {
+                              "Fn::If": [
```

Comment:
Placing your `chain-script` and `make-app` into the same `configSet` would eliminate the need for this condition-block.